src/main/webapp/resources/celJS/initCelements.js

```diff
@@ -132,13 +132,35 @@
     };
   }
 
+  /**
+   * celOnBeforeLoadListener
+   */
   window.celOnBeforeLoadListenerArray = [];
 
+/**
+ * @deprecated celAddOnBeforeLoad is deprecated since open-celements 5.4 / January 2022.
+ * Instead register a listener on "DOMContentLoaded"
+ */
   window.celAddOnBeforeLoadListener = function(listenerFunc) {
+    console.warn('celAddOnBeforeLoad is deprecated since open-celements 5.4 / January 2022.'
+    + ' Instead register a listener on "DOMContentLoaded"', listenerFunc);
     celOnBeforeLoadListenerArray.push(listenerFunc);
   };
 
-  window.celOnFinishHeaderListenerArray = [];
+document.addEventListener('DOMContentLoaded', function() {
+  if(celOnBeforeLoadListenerArray
+      && (typeof celOnBeforeLoadListenerArray !== 'undefined')) {
```

Comment:
a var evaluating to truthy cannot possibly be undefined. also truthy evaluation would fail if it were undefined at this point.
```suggestion
  if(typeof celOnBeforeLoadListenerArray !== 'undefined')) {
```